### ISSUE TITLE
facets: make linked axes dependent upon the direction keyword

### ISSUE THREAD

**Issue by rapus95** [JuliaPlots/AlgebraOfGraphics.jl#335]
`visual(BarPlot, direction=:x)` links the x-axes as usual, which is wrong in this case.

**Comment by rapus95** [comment]
As I just saw, it looks like AoG cannot handle the direction keyword at all since the axes are named as if the axes weren't flipped

**Comment by piever** [comment]
Yes, it's waiting for https://github.com/JuliaPlots/Makie.jl/issues/926 to handle it without too many contorsions, but it's good to keep in mind that axes linking should also take this into account.

**Comment by rapus95** [comment]
until then, is there any easy way to flip the plot direction aside of manually drawing it with Makie directly?

**Comment by piever** [comment]
You can probably hack your way around it with something like

```julia
@recipe(FlippedBarPlot) do scene
    return default_theme(scene, BarPlot)
end

Makie.convert_arguments(::Type{<:FlippedBarPlot}, x, y) = PlotSpec{BarPlot}(y, x, direction=:x)

plt = data((x=rand(3), y=["a", "b","c"])) * mapping(:x, :y) * visual(FlippedBarPlot)
```

where the association variable -> axis is preserved in the recipe `FlippedBarPlot`, so everything should work

**Comment by jfb-h** [comment]
> You can probably hack your way around it with something like

I also just encountered this issue and tried out the hack, but it seems it doesn't actualy use the x values but instead just integers 1:length(y). I'm not that familiar with the Makie recipe system, is that just due to a typo or something more involved?

**Comment by jfb-h** [comment]
Ah just saw the other issue, sorry for the noise. Feel free to ignore!
